### DISCUSSION TITLE
Allowed Image Patterns label

### DISCUSSION THREAD

**Discussion by rak-phillip** [rancher-sandbox/rancher-desktop#3699] [General]
We can provide more context in the label for the Allowed Image Patterns preference so that the meaning of this preference is clear.

Some  proposed options from out-of-band discussions:

- (Define | Specify) a list of patterns to (restrict | limit) images pulled from a registry
- (Specify | Define) patterns for pulling images from a registry
- (Limit | Restrict | Define | Specify) images that can be pulled from a registry 
- Define regex patterns for images that can be pulled from all registries

We must communicate that this is an allow list and that specifying image patterns is only relevant when the checkbox is ticked.

### Motivation

We can provide additional context in labels for other preferences in Rancher Desktop, for example:

- Administrative Access: Allow Rancher Desktop to acquire administrative credentials (sudo access)
- Automatic Updates: Check for updates automatically
- Statistics: Allow collection of anonymous statistics to help us improve Rancher Desktop

**Comment by jandubois** [comment]
Notes:

* the restriction applies to pushing of images as well
* we will replace regexp with wildcard patterns

Alternative:

We could change the `Enabled` checkbox to 2 radio buttons:

```
Allowed Images
(*) All  ( ) Name must match at least one pattern

Patterns:
docker.io/library/*
...
```
